src/ports/types.ts

```diff
@@ -0,0 +1,47 @@
+export type Undefined = null | undefined | 0 | '';
```

Comment:
Please don't do that. Undefined should be clearly defined. `null` or `undefined` depending on use case. No type coercion